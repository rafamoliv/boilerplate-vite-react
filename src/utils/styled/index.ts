import { css } from 'styled-components'

export const getColors = (prop1, prop2) => {
  return ({ theme }) => css`
    ${theme.colors[prop1][prop2]};
  `
}
