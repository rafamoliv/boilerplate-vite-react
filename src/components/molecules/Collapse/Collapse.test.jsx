import { expect, it, vi } from 'vitest'

import { render, screen, userEvent } from '@/utils/tests'

import { Collapse } from '.'

const props = {
  onTooggle: vi.fn(),
  children: 'Ol√° mundo'
}

describe('<Collapse />', () => {
  describe('Component Render', () => {
    it('render element', () => {
      const { container } = render(<Collapse {...props} />)
      const button = screen.getByRole('button', '/Mostrar/i')
      expect(button).toBeInTheDocument()
      expect(container.firstChild).toMatchSnapshot()
    })
  })

  describe('Events', () => {
    it('Click and show element', async () => {
      render(<Collapse {...props} />)
      const button = screen.getByRole('button', '/Mostrar/i')
      await userEvent.click(button)

      const content = screen.getByText(props.children)
      expect(content).toBeInTheDocument()
    })
  })
})
