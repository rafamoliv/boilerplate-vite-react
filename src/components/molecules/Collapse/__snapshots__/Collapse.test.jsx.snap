// Vitest Snapshot v1

exports[`<Collapse /> > Component Render > render element 1`] = `
.c1 {
  background: #f231a5;
  padding: 0.5rem 1rem;
  margin: 1em;
  border: 2px solid #b81979;
  border-radius: 3px;
  color: #fff;
}

.c0 {
  border: solid 1px #f231a5;
}

@media (min-width:48rem) {
  .c0 {
    font-size: 1.5rem;
  }
}

<div
  class="c0"
>
  <button
    class="c1"
  >
    Mostrar
  </button>
</div>
`;
